src/Quadrature.cpp

```diff
@@ -0,0 +1,1305 @@
+
+  if (_num_polar/2 * _num_azim/4 != num_azim_times_polar)
+    log_printf(ERROR, "Unable to set %d polar weights for Quadrature "
+               "with %d polar angles and %d azimuthal angles"
+               " in each octant",
+               num_azim_times_polar, _num_polar/2, _num_azim/4);
+
+  /* Deallocate memory if it was allocated previously */
+  if (_polar_weights != NULL) {
+    for (int a=0; a < _num_azim/2; a++)
+      delete [] _polar_weights[a];
+    delete [] _polar_weights;
+  }
+
+  /* Initialize memory for arrays */
+  _polar_weights = new FP_PRECISION*[_num_azim/2];
+  for (int i=0; i < _num_azim/2; i++)
+    _polar_weights[i] = new FP_PRECISION[_num_polar];
+
+  /* Extract polar weights from user input */
+  int ap=0;
+  for (int a=0; a < _num_azim/4; a++) {
+    for (int p=0; p < _num_polar/2; p++) {
+      if (weights[ap] < 0. || weights[ap] > M_PI_2)
+        log_printf(ERROR, "Unable to polar weight to %f which is "
+                   "not in the range [0,PI/2]", weights[ap]);
+
+      setPolarValues(_polar_weights, a, p, weights[ap]);
+      ap++;
+    }
+  }
+}
+
+
+/**
+ * @brief Sets the polar angle for the given indexes
+ * @param theta the value of the polar angle to be set
+ * @param azim the azimuthal index of the angle of interest
+ * @param polar the polar index of the angle of interest
+ */
+void Quadrature::setTheta(double theta, int azim, int polar) {
+
+  if (theta <= 0.0 || theta >= M_PI_2)
+    log_printf(ERROR, "Unable to set theta for azim = %d and polar = %d "
+               "to %f which is not in the range (0.0, PI/2)", azim, polar,
+               theta);
+
+  else if (azim >= _num_azim/4)
+    log_printf(ERROR, "Unable to set theta for azim = %d and polar = %d "
+               "since azim is not in the range (0, _num_azim/4)", azim, polar);
+
+  else if (polar >= _num_polar/2)
+    log_printf(ERROR, "Unable to set theta for azim = %d and polar = %d "
+               "since polar is not in the range (0, _num_polar/2)",
+               azim, polar);
+
+
+  if (_thetas == NULL) {
+    _thetas = new double*[_num_azim/2];
+    for (int i=0; i < _num_azim/2; i++)
+      _thetas[i] = new double[_num_polar];
+  }
+
+  _thetas[azim][polar] = theta;
+  _thetas[_num_azim/2 - azim - 1][polar] = theta;
+  _thetas[azim][_num_polar - polar - 1] = M_PI - theta;
+  _thetas[_num_azim/2 - azim - 1][_num_polar - polar - 1] = M_PI - theta;
+}
+
+
+/**
+ * @brief Sets the azimuthal angle for the given index
+ * @param phi the value of the azimuthal angle to be set
+ * @param azim the azimuthal index
+ */
+void Quadrature::setPhi(double phi, int azim) {
+
+  if (phi <= 0.0 || phi >= M_PI_2)
+    log_printf(ERROR, "Unable to set phi for azim = %d to %f which is not "
+               "in the range (0.0, PI/2)", azim, phi);
+
+  else if (azim >= _num_azim/4)
+    log_printf(ERROR, "Unable to set phi for azim = %d since azim is not in"
+               " the range (0, _num_azim/4)", azim);
+
+  else if (_phis == NULL)
```

Comment:
Although it doesn't actually matter here, I think it is better to make this memory allocation block a standalone `if` rather than an `else if` since it doesn't logically connect to the error checking blocks above.